### PR TITLE
Milestone 2: user_interface

### PR DESCRIPTION
**Pull request by pastorp3**
- In this milestone, we set the puts for the user and the gets to collect the data

- We create a class to display the board and user interface

### CODE REVIEW

File: bin/main [hunk 1]
```diff
@@ -1,2 +1,32 @@
...
+
+  def user_input
+    puts 'Please enter value between 1-9'
+    user_input = Integer(gets.chomp)
+    user_input -= 1
+    chosen_index = user_input
+    puts chosen_index
+    display_board
+  end
+end
```
Comment (kitpao) [inline bin/main:26]: - [ ] Use of classes is encouraged 💯 However, if you are going to use them, put them into your lib folder in a file named after the class. This helps to keep your code organized. Don't forget to use require_relative to link the files

File: bin/main [hunk 2]
```diff
@@ -1,2 +1,32 @@
...
+    puts '----------'
+    puts "#{@board[6]} | #{@board[7]} | #{@board[8]} "
+    puts "\n"
+  end
+
+  def user_input
+    puts 'Please enter value between 1-9'
+    user_input = Integer(gets.chomp)
+    user_input -= 1
+    chosen_index = user_input
```
Comment (kitpao) [inline bin/main:22]: - [ ] Use validations whenever you ask something from the user, what will happen if the user press "F" or "25"? **Hint:** You can use loops or ruby exceptions (begin/rescue/else/ensure)

File: bin/main [hunk 3]
```diff
@@ -1,2 +1,32 @@
...
+    puts chosen_index
+    display_board
+  end
+end
+
+game = Game.new
+puts '----------Welcome------------'
+puts '--------Tic Tac Toe----------'
+game.display_board
+game.user_input
```
Comment (kitpao) [inline bin/main:32]: - [ ] Remember there will be more kinds of inputs like the name of the player also. So try to use more descriptive names for your methods. Additionally (and the following is optional 👉 ) you can display the rules by pressing "i" for example, or ask the user what will their character be "X" or "O", as you wish

Comment (kitpao) [inline bin/main:32]: - [ ] Your flow is still incomplete, for all the parts where you need logic, use inline comments (example: `# here it checks for a winner`) and then use puts (`# if player one won, then:` `puts "Player 1 won the game"`)
This is your sketch of how the game will progress

**Comment by pastorp3** [comment]
- add line comments for each method

- add a file to lib folder with part of logic with comments

File: bin/main (hunk 3 above)
Comment (happiguru) [reply bin/main]: Updated with complete code

File: bin/main (hunk 2 above)
Comment (happiguru) [reply bin/main]: Used validation

File: bin/main (hunk 1 above)
Comment (happiguru) [reply bin/main]: We made use of require_relative

**Comment by pastorp3** [comment]
- add loop while

- show player turn

- code not working at all without logic

**Review by peterrobert — APPROVED** [review]
### APPROVED :100:  ON TO THE NEXT MILESTONE.

Great work implementing all the required changes.

**Pull request merged**